### PR TITLE
chore: bump dependencies

### PR DESCRIPTION
**Pull request by bradzacher**
`lint-staged` is the only one I didn't bump, because they dropped support for node 8, and we still support it. I think most people work in here on node 10/12, but I figured that it's probably best to just wait till we drop support too.

### CODE REVIEW

**Comment by bradzacher** [comment]
Ahhh phooey.
I thought my PR (https://github.com/eslint/eslint/pull/12715) had been released, but it was merged a few days late, so it's going to be in the 7.0 release instead.

I'll have to revert all of my changes, or bump to eslint 7.0.0-alpha.0
https://github.com/eslint/eslint/releases/tag/v7.0.0-alpha.0

File: packages/eslint-plugin/package.json [hunk 1]
```diff
@@ -42,17 +42,16 @@
...
   "dependencies": {
     "@typescript-eslint/experimental-utils": "2.18.0",
-    "eslint-utils": "^1.4.3",
+    "eslint-utils": "^2.0.0",
     "functional-red-black-tree": "^1.0.1",
     "regexpp": "^3.0.0",
     "tsutils": "^3.17.1"
   },
   "devDependencies": {
-    "@types/json-schema": "^7.0.3",
```
Comment (armano2) [inline packages/eslint-plugin/package.json]: nit

File: packages/eslint-plugin/tests/rules/quotes.test.ts [hunk 2]
```diff
@@ -10,23 +10,26 @@ const ruleTester = new RuleTester({
...
-// eslint-disable-next-line @typescript-eslint/no-explicit-any
-const useBacktick: any = {
-  message: 'Strings must use backtick.',
-};
+const useBacktick = {
+  messageId: 'wrongQuotes',
+  data: {
+    description: 'backtick',
+  },
+} as const;
```
Comment (armano2) [inline packages/eslint-plugin/tests/rules/quotes.test.ts]: 👍

File: packages/eslint-plugin/package.json (hunk 1 above)
Comment (bradzacher) [reply packages/eslint-plugin/package.json]: we don't consume the types via the plugin - we consume them via `experimental-utils`.
https://github.com/typescript-eslint/typescript-eslint/blob/b24fbe8790b540998e4085174251fb4d61bf96b0/packages/eslint-plugin/src/rules/naming-convention.ts#L3

I did this on purpose to make sure that the versions were the same. IIRC I ran into a problem once when `experimental-utils` had one version and `eslint-plugin` had another, so there were two copies installed which caused weirdness.

Comment (armano2) [reply packages/eslint-plugin/package.json]: you are correct:

```
'json-schema' should be listed in the project's dependencies. Run 'npm i -S json-schema' to add it  import/no-extraneous-dependencies
```

Comment (bradzacher) [reply packages/eslint-plugin/package.json]: actually yeah! That was the entire reason for exposing it from `experimental-utils`.
You have to explicitly import the types, but it's a type-only module, so our lint rule failed.
By exporting it from the utils package, I could consolidate the lint disable into a single place.

**Comment by armano2** [comment]
seems that tests did fail again on integration tests:

```
Failed to load plugin '@typescript-eslint' declared in '--config': createRequire is not a function
typescript-and-tslint-plugins-together    | Referenced from: /usr/linked/.eslintrc.yml
typescript-and-tslint-plugins-together    | FAIL ./test.js
typescript-and-tslint-plugins-together    |   ● Test suite failed to run
typescript-and-tslint-plugins-together    | 
typescript-and-tslint-plugins-together    |     Cannot find module './lint-output.json' from 'test.js'
typescript-and-tslint-plugins-together    | 
typescript-and-tslint-plugins-together    |     > 1 | const actualLintOutput = require('./lint-output.json');
typescript-and-tslint-plugins-together    |         |                                  ^
typescript-and-tslint-plugins-together    |       2 |  
typescript-and-tslint-plugins-together    |       3 |  test('it should produce the expected lint ouput', () => {
typescript-and-tslint-plugins-together    |       4 |    expect(actualLintOutput).toMatchSnapshot();
typescript-and-tslint-plugins-together    | 
typescript-and-tslint-plugins-together    |       at Resolver.resolveModule (node_modules/jest-resolve/build/index.js:276:11)
typescript-and-tslint-plugins-together    |       at Object.<anonymous> (test.js:1:34)

```

**Comment by bradzacher** [comment]
weird... the integration tests aren't working at all..

**Comment by bradzacher** [comment]
This needs work.

**Pull request closed without merging**